src/Microsoft.NetCore.Analyzers/Core/Tasks/DoNotCreateTaskCompletionSourceWithWrongArguments.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Tasks
+{
+    /// <summary>
+    /// RS1027: Do not create TaskCompletionSource with wrong arguments
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotCreateTaskCompletionSourceWithWrongArgumentsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "RS1027";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotCreateTaskCompletionSourceWithWrongArgumentsTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotCreateTaskCompletionSourceWithWrongArgumentsMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotCreateTaskCompletionSourceWithWrongArgumentsDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.MicrosoftCodeAnalysisCorrectness,
```

Comment:
Incorrect category - this category is reserved for `Microsoft.CodeAnalysis.Analyzers`

Reply:
Please select the category from https://github.com/dotnet/roslyn-analyzers/blob/6d265703c1ccd10176dc77e768412cf1422f9153/src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt#L5-L15

Reply:
> Please select the category from

So we consider these new rules that were never "FxCop" to still be "FxCop"?  Maybe we should update that?

Reply:
Yes, I think we will soon replace all FxCop terminology in the repo with something different, say Managed code analysis (CA) rules.

Reply:
Updated category to DiagnosticCategory.Usage and updated RuleID to CA2247 in https://github.com/dotnet/roslyn-analyzers/pull/3059/commits/fb64a67c4c645d26b50037b4b58f1a50206aa24d